src/formatDate.js

```diff
@@ -8,7 +8,40 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let year, month, day;
+  const parts = date.split(fromFormat[3]);
+
+  fromFormat.slice(0, 3).forEach((format, index) => {
+    if (format === 'YYYY' || format === 'YY') {
+      year = parts[index];
+    } else if (format === 'MM') {
+      month = parts[index];
+    } else if (format === 'DD') {
+      day = parts[index];
+    }
+  });
+
+  if (year.length === 4 && toFormat.includes('YY')) {
+    year = year.slice(2);
+  } else if (year.length === 2 && toFormat.includes('YYYY')) {
+    year = (year < 30 ? '20' : '19') + year;
```

Comment:
The logic for determining the century for two-digit years assumes that any year less than 30 belongs to the 2000s and any year 30 or greater belongs to the 1900s. This assumption may not be valid for all use cases and could lead to incorrect results for dates outside this range.